pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/MaxAggregationFunction.java

```diff
@@ -28,13 +28,19 @@
 import org.apache.pinot.core.query.aggregation.groupby.DoubleGroupByResultHolder;
 import org.apache.pinot.core.query.aggregation.groupby.GroupByResultHolder;
 import org.apache.pinot.segment.spi.AggregationFunctionType;
+import org.roaringbitmap.RoaringBitmap;
 
 
 public class MaxAggregationFunction extends BaseSingleInputAggregationFunction<Double, Double> {
   private static final double DEFAULT_INITIAL_VALUE = Double.NEGATIVE_INFINITY;
+  private final boolean _nullHandlingEnabled;
 
-  public MaxAggregationFunction(ExpressionContext expression) {
+  // stores id of the groupKey where the corresponding value is null.
+  private Integer _groupKeyForNullValue = null;
```

Comment:
Do we need to handle the null group in the aggregation function? It will make aggregation function hard to implement. I think we should be able to identify and skip the null groups in the group-by executor

Reply:
The default flow `aggregateGroupBySV()` cannot differentiate between default values that are null values are default values that are genuine values. Also, assuming you want to compute MIN and MAX aggregation functions on your data which has default values, you cannot choose a default value that does not impact the computation of both MIN and MAX.

Reply:
Understood. We definitely need to track the null values in the aggregated column, but do we need to track the `_groupKeyForNullValue` in the aggregation function as well? I think we already track it in the group-by executor, which should be enough

Reply:
@Jackie-Jiang The problem with tracking `groupKeyForNullValue` in `GroupByExecutor` is the following:
- In `MaxAggregationFunction`, `DEFAULT_INITIAL_VALUE` is `Double.NEGATIVE_INFINITY`. `aggregateGroupBySV()` calls `setValueForKey` only if the value > `DEFAULT_INITIAL_VALUE` (or the latest set value). `DEFAULT_INITIAL_VALUE` could be a valid non-null value or a null value (e.g. dimension column). If `nullHandlingEnabled` is set to true, we should not run this check (value > `DEFAULT_INITIAL_VALUE`). In summary, if `nullHandlingEnabled` is set to true, the whole range of values in a data type is a valid range including `Double.NEGATIVE_INFINITY`.
- The same applies to `MinAggregationFunction`, and the `DEFAULT_VALUE` = `Double.POSITIVE_INFINITY`.

Reply:
I'm still confused. In `aggregateGroupBySVNullHandlingEnabled()`, the `nullBitmap` is for the aggregated value instead of the group-by column. When the aggregated value is null, why do we set the `_groupKeyForNullValue`? There can be multiple aggregated value to be null, but with different group key.

Reply:
Thanks for the input. Addressed.